crypto/dso/dso_dlfcn.c

```diff
@@ -308,6 +308,71 @@ static int dladdr(void *address, Dl_info *dl)
 }
 # endif                         /* __sgi */
 
+# ifdef _AIX
+/*-
+ * See IBM's AIX Version 7.2, Technical Reference:
+ *  Base Operating System and Extensions, Volume 1 and 2
+ *  https://www.ibm.com/support/knowledgecenter/ssw_aix_72/com.ibm.aix.base/technicalreferences.htm
+ */
+#  include <sys/ldr.h>
+#  include <errno.h>
+/* ~ 64 * (sizeof(struct ld_info) + _XOPEN_PATH_MAX + _XOPEN_NAME_MAX) */
+#  define DLFCN_LDINFO_SIZE 86976
+typedef struct Dl_info {
+    const char *dli_fname;
+} Dl_info;
+/* NOTE: when looking up a function symbol, the caller of DSO_...byaddr() /
+ *  OPENSSL_atexit() might need to consider the Pointer Glue virtual address
+ *  handling on AIX/PowerPC (aka De-Virtualization; related to the ptrgl
+ *  compiler flags). The real address of a function can be retrieved with:
+ *  (void*)*((ulong*)function)
```

Comment:
I kind of disagree with this wording. Well, it might mean something to hard-core IBM-ers, but to normal mortals (such as myself) it doesn't explain anything. Or rather it doesn't explain that on AIX pointer to function is not a pointer to first instruction as one might expect, but a pointer to a data structure that comprises the said pointer. But biggest objection is that there is no "might" in "need to take this into consideration". DSO_...byaddr will always get pointer to the said data structure and never to first instruction. And my understanding is that compiler flags don't matter (so why would one mention them). Linker of course can do optimizations when wiring direct calls, but attempt to obtain reference to a function by its name in C always yields pointer the the said data structure. Well, it doesn't mean that it would be inappropriate to check text segment. Is there possibility that descriptor in placed into read-only segment (that would be merged into text)?

And while you're at it. Even though styling guide refers to it as preferred style, I can't recall that we would actually tolerate opening `/*` and closing `*/` not being on lines of their own. Not in new code. This is not the only case in suggested code.

Reply:
I have been shewing on your comment for a while now. You are right, this doesn't belong to this place, so I'll remove it.
I'll edit the comments to have `/*` and `*/` on separate lines. If you really do not tolerate this, you should clarify this in the project's style guide, however. Chapter 8 doesn't even mention it, currently. I am used to compact comments easily fitting into IDE popups.

Reply:
Just in case it wasn't obvious. "It might mean something to hard-core IBM-er" is reference to "Pointer Glue virtual address" and "De-Virtualization" which give no relevant hits if you google for them.

As for styling commentary. Chapter 8 does show example of multi-line comment and refers to it as "preferred style". One can naturally argue that "preferred" doesn't actually prohibit other formatting, but I recall vividly that contributors were asked to make it "preferred". Sorry about confusion...